regparser/grammar/interpretation_headers.py

```diff
@@ -25,10 +25,16 @@
     + unified.depth1_p
 )
 
-
+# This matches an appendix name in an appendix header. Here we'll match
+# something with a dash in the appendix name (i.e. AA-1) but we'll
+# remove the dash. The effect of this is that, for label purposes only,
+# the appendix becomes known as 'AA1', and therefore we don't have weird
+# label collisions with a node labeled '1' underneath the appendix.
 appendix = (
     atomic.appendix_marker.copy().leaveWhitespace()
-    + (unified.appendix_with_section | atomic.appendix )
+    + Regex(r"[A-Z]+-?[0-9]*\b").setResultsName("appendix").setParseAction(
```

Comment:
This is also why I say this may complicate internal citations to these "special" appendices.